src/main/java/bookstore/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,35 @@
+package bookstore.repository;
+
+import bookstore.entity.Book;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.hibernate.SessionFactory;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@RequiredArgsConstructor
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory factory;
+
+    @Override
+    public Book save(Book book) {
+        try {
+            factory.inTransaction(e -> e.persist(book));
+        } catch (Exception e) {
+            throw new RuntimeException("Save operation was unsuccessful: ", e);
```

Comment:
It's bad practice to throw RuntimeException. Let's create custom DataProcessingException
The message could be "Can't save book to DB"